src/__tests__/helpers.js

```diff
@@ -43,3 +50,19 @@ describe('query container validation throws when validation fails', () => {
     )
   })
 })
+
+test('should always use realTimers before using callback', () => {
+  const originalSetTimeout = globalObj.setTimeout
+
+  jest.useFakeTimers('legacy')
+  runWithRealTimers(() => {
+    expect(originalSetTimeout).toEqual(globalObj.setTimeout)
+  })
+
+  jest.useRealTimers()
+
+  jest.useFakeTimers('modern')
```

Comment:
Because of these tests, we'll need an `afterEach` that does `jest.useRealTimers()` to keep this isolated.

Reply:
totally right I forgot it